nservicebus/hosting/publishing-from-web-applications.md

```diff
@@ -4,67 +4,58 @@ summary: Guidelines for publishing messages from within web applications
+ 1. **Transactions and consistency** - [Events should announce that something has *already* happened](/nservicebus/messaging/messages-events-commands.md). Many times this involves making changes to a database, then publishing the result. But HTTP is inherently unreliable and does not have built-in retries. If an exception occurs before the event is published, the only opportunity to publish that event may be lost. In these circumstances, it's generally better to send a command with the payload of the HTTP request, and have another endpoint process the command. Doing this takes advantage of NServiceBus's [recoverability capabilities](/nservicebus/recoverability/).
  1. **Resource optimization** - The most precious resource in a web server is a request processing thread. When these are exhausted, the server can no longer handle additional load and must be scaled out. Complex database updates (possibly involving the Distributed Transaction Coordinator) can block these request threads while waiting for I/O to and from the database. In these situations, it's more efficient to send a command to a back-end processor. Typically this allows for better scalability with fewer server resources.
  1. **Web application scale-out** - Because events can only be published from one logical publisher, it can be problematic to scale out a web application that is publishing events. More on this below.
 
 Given these facts, conventional wisdom suggests that in the context of a web application, it is better to only send commands to a back-end service endpoint, which can then publish a similar event.
 
-But this is not always the case:
-
- * NServiceBus version 4.0 introduced the ability to have [transports other than MSMQ](/transports/). Not all of the transports handle publish/subscribe mechanics in the same way as MSMQ - some even support it natively.
- * Cloud transports generally have an associated cost per interaction; sending a message unnecessarily before publishing costs money to operate.
- * In NServiceBus versions 4.0 and 5.0, the process of creating a send-only endpoint (which is still capable of publishing messages, but does not have an input queue or process incoming messages) was made much easier.
-
+But this is not always the case. Not all [transports](/transports/) handle publish/subscribe mechanics in the same way - many support it natively. Also, cloud transports generally have an associated cost per interaction; sending a message unnecessarily before publishing costs money to operate.
```

Comment:
Can we maybe link here to https://docs.particular.net/transports/selecting in some way?